sdk/go/pulumi/stack_reference.go

```diff
@@ -51,6 +51,45 @@ func (s *StackReference) GetOutput(name StringInput) AnyOutput {
 		}).(AnyOutput)
 }
 
+// StackReferenceOutputDetails holds a stack output value.
+// At most one of the Value and SecretValue fields will be set.
+//
+// You can build a StackReferenceOutputDetails with
+// the [StackReference.GetOutputDetails] method.
+type StackReferenceOutputDetails struct {
+	// Output value returned by the StackReference.
+	// This field is nil if the value is a secret
+	// or it does not exist.
+	Value interface{}
```

Comment:
Horrible corner case that I don't think we need to worry about, but I think an output can be "null" itself. But I imagine hardly anything in Go handles that correctly anyway.

Reply:
Good point. Yeah, if we get to a point where we want to support `null` in the model, it might be possible to introduce a sentinel value `pulumi.Null`. Although I don't think this is likely for the top-level object.